packages/core/src/odata/common/selectable/collection-field.ts

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import { EntityBase } from '../entity';
+import { Constructable } from '../constructable';
+import { SelectableEdmTypeField } from './edm-type-field';
+import { Field } from './field';
+import { ComplexTypeField } from './complex-type-field';
+import { SimpleTypeFields } from './simple-type-fields';
+
+export class CollectionField<EntityT extends EntityBase> extends Field<EntityT>
+  implements SelectableEdmTypeField {
+  readonly selectable: true;
+  constructor(
+    readonly _fieldName: string,
+    readonly _entityConstructor: Constructable<EntityT>,
+    // TODO: for simple type fields this can be improved to just use the edm type
+    // for complex types this is a little more complex due to the nature of complex types and should be revisted latest for v2.0
+    readonly _fieldType: SimpleTypeFields<EntityT> | ComplexTypeField<EntityT>
```

Comment:
General question, I know this is the way before. But should the collection type have two generic parameters: one for the entity and one for the type of the internal data type?

Reply:
Perhaps a thing we could als add a ticket for?

Reply:
I agree, we actually said, this would be done in a follow up, not sure what happened here. @jjtang1985 do we already have a follow up for this?